File: 5.implementation/VT/redsun-service/src/main/java/com/redsun/service/service/PriorityService.java
package com.redsun.service.service;

import com.redsun.service.entities.Priority;
import com.redsun.service.entities.Result;

public interface PriorityService {

	//select on page
	Result getPriority(int client_id, String name, int itemsPerPage, int pageNo);

	//select all priorities
	Result listAllPriorities(int clientId);

	//insert
	Result insert(Priority priority);

	//update
	Result update(Priority priority);

	//delete
	Result delete(Integer id);

	//get by id
	Result getById(Integer id);

}

File: 5.implementation/NT/redsun/src/main/webapp/static/js/services/priorityService.js
app.factory('priorityService', function($http, $rootScope) {
	// Might use a resource here that returns a JSON array
	return {
		getPriorities : function($scope, $rootScope, $http, pageNo) {
			var url = $rootScope.makeGetURL("/priority/" + $scope.name + "/" + $scope.itemsPerPage + "/"+ pageNo);
			$http.get(url).then(function(response) {
				if (response.data) {
					$scope.data = angular.fromJson(response.data);
				}
				$scope.$broadcast('getPrioritiesAlready');
			});

			return true;
		}

		,
		getById : function($scope, $rootScope, $http) {
			var url = $rootScope.makeGetURL("/priority/getbyid/" + $scope.priorityId);
			$http.get(url).then(function(response) {
				response = angular.fromJson(response);
				$scope.priority = response.data.result.priorities;
				$scope.$broadcast('getPriorityDetailAlready');
			})
			return false;
		}
		// Update.
		,
		update : function($scope, $rootScope, $http) {
			var urlServer = $rootScope.makePostURL("/priority/update/" +  $scope.priorityId);
			$http.put(urlServer, $scope.priority).then(function(response) {
				if(response.data){
					$scope.data = angular.fromJson(response.data);
				}
			});
			return true;
		}
		,
		delete : function($scope, $rootScope, $http) {
			var urlServer = $rootScope.makePostURL("/priority/delete/" +  $scope.priorityId);
			return $http.delete(urlServer);
		}
		

	};
});